lib/utils/filter_samples.py

```diff
@@ -130,9 +137,24 @@ def _format_and_validate_filters(self, parsed_filters):
                 message = "Unable to resolve metadata fields or prefix metadata fields: " + \
                           ", ".join(list(key_set))
                 raise ValueError(message)
+
+        # check if there are any bad uncontrolled fields
```

Comment:
totally a "your opinion" thing but it may be worth pulling this out into a named function (validate_uc_fields?) for clarity then just
```python
if len(uc_filters):
    validate_uc_fields(uc_filters, samples, token)
```